Modules/_sqlite/connection.c

```diff
@@ -1225,6 +1251,11 @@ int pysqlite_check_thread(pysqlite_Connection* self)
 
 static PyObject* pysqlite_connection_get_isolation_level(pysqlite_Connection* self, void* unused)
 {
+    if (!self || !self->isolation_level) {
+        PyErr_Format(pysqlite_ProgrammingError,
+                    "Object is null or isolation_level is uninitialized.");
+        return 0;
+    }
```

Comment:
If `self->initialized` is only set if `pysqlite_connection_init` was successful, this can be reduced to:

```suggestion
    if (!pysqlite_check_connection(self)) {
        return NULL;
    }
```